### ISSUE TITLE
Styling API

### ISSUE THREAD

**Issue by harshzalavadiya** [megahertz/react-simple-wysiwyg#4]
This project looks amazing fulfills minimal editor scenario perfactly

Since I believe that this project wants to keep component minimal but having ability to modify styles might be nice to have feature.

![image](https://user-images.githubusercontent.com/5774849/108232555-e8ba6800-7168-11eb-9140-73fd25f77113.png)

#### Targetable Classes
having classes on components like `rcw-toolbar` etc so it will be easy to target styling overrides

#### Inlined Styles 
Inlined styled are hard to override since we have to put `!important` everywhere it also introduces style repeations everywhere
1. seprate minified and bundled css file
2. adding tiny version of styled-component replacing runtime like [goober](https://github.com/cristianbote/goober)

I'll be happy to provide a PR if you are intrested

**Comment by megahertz** [comment]
Thank you for the suggestion. Yes, I think it's a nice idea. I would prefer to use className pointing to the style injected to head. I think this way allows to keep minimal size and doesn't require to care about separated css file. What do you think? I would appreciate if you could make a PR, since I have no time for that till March.

**Comment by harshzalavadiya** [comment]
@megahertz that's great idea, I'll try to work on this over weekends

**Comment by harshzalavadiya** [comment]
@megahertz

After looking at project there was changes might be necessary but since they are quite large I thought of asking you first before working on it

### Move to TSDX
- I was able to create PoC but only using [tsdx](https://github.com/formium/tsdx)(behind the scene also uses rollup)
- I tried with current rollup config but no luck
- there was no build size change when I just moved from plain rollup to tsdx based see below for output

```txt
dist/react-simple-wysiwyg.cjs.production.min.js
Size limit: 10 KB
Size:       3.42 KB with all dependencies, minified and gzipped

dist/react-simple-wysiwyg.esm.js
Size limit: 10 KB
Size:       3.49 KB with all dependencies, minified and gzipped
```
[Bundlephobia of current package](https://bundlephobia.com/result?p=react-simple-wysiwyg@1.0.2)

### Remove direct `style` API since classes can be easily overridden
- will it be okay if we remove styles prop from editor since they can be easily overwritten using user's preffered provided target `className` on each prop?, this way we can remove [deepMerge](https://github.com/megahertz/react-simple-wysiwyg/blob/bfea942c557cd0025af3c16f208d3dda2109a6ed/src/Editor.tsx#L63) code and will be able to achieve same thing
- this way we can avoid extra complixity and reduce bundle size even more

**Comment by megahertz** [comment]
Thanks @harshzalavadiya, it looks interesting

A few notes:
- I would prefer to configure bundling directly without tools like tsdx
- I would prefer to have a single css file which is injected only once.

**Comment by harshzalavadiya** [comment]
@megahertz 
> I would prefer to configure bundling directly without tools like tsdx
Sure, I'll try with that if possible

> I would prefer to have a single css file which is injected only once.
The reason I thought of multiple styles so components can be imported like lego blocks and still work. but sure can put everything in a root editor so this will be applied as expected
